### DISCUSSION TITLE
Work on windows and linux ?

### DISCUSSION THREAD

**Discussion by bepolymathe** [freeplane/freeplane#477] [Help]
Hi, 

What is the best way to work with Freeplane on two computers? A pro computer under windows. A personal computer under linux (Fedora). 

I would obviously like to keep all my configuration (and modifications) on both computers. I would also like to synchronize all the data imported in the maps (attached files, etc). 

Would a simple synchronization solution with Nextcloud or other be viable?

Thanks in advance.

**Comment by macmarrum** [comment]
A great idea 👏 

> Would a simple synchronization solution with Nextcloud or other be viable?

I have not tried it, but most probably yes. This is assuming that only one instance of Freeplane will be running at a time (either pro or personal), and that `Preferences->Environment->Hyperlink Types->Links` will be set to `Relative`.

To auto-sync preferences, you will need to specify a custom `user dir`, so that preferences are located in the synchronized folder: 
* `-Dorg.freeplane.userfpdir=$userfpdir` – default on Linux, as seen in `freeplane.sh`
* `-Dorg.freeplane.userfpdir="%APPDATA%\\Freeplane"` – default on Windows, as seen in `freeplane.bat`

and you will need to start Freeplane using your customized scripts.

The file paths on Linux and Windows use a different format, therefore `Most recent maps` will probably cause some issues or simply not work.

I am curious about the end result. Please report back in any case 😄

**Reply by bepolymathe** [reply]
The potential problems you point out are the ones I was thinking of. The portable version of Freeplane would not solve this problem either. 
I will do some tests.

**Comment by ofossum** [comment]
In addition to the thoughts of separate user and data-dirs from @macmarrum : 
try [Syncthing](https://syncthing.net/) for synchronizing the different folders. 
It works like a charm for me since a long time.
Both computers have to be online for syncing but you can easily sync folders in different places on different systems with it.
